Cabal/Distribution/Simple/InstallDirs.hs

```diff
@@ -286,7 +286,7 @@ absoluteInstallDirs :: PackageIdentifier
                     -> InstallDirs FilePath
 absoluteInstallDirs pkgId libname compilerId copydest platform dirs =
     (case copydest of
-       CopyTo destdir -> fmap ((destdir </>) . dropDrive)
+       CopyTo destdir -> fmap ((destdir </>) . takeBaseName)
```

Comment:
I don't think this is right. Here you take `InstallDirs` and substitute all absolute paths by `destdir </> basename absPath`. E.g. if `libdir` was `$prefix/lib/$abi/$libname`, you make it `$destdir/$libname`.

Reply:
Though it seems to me that it's underspecified what `copy --destdir` should mean. I'd expect the value of `--destdir` to replace `$prefix`, but apparently it just gets prepended to the full absolute path including `$prefix` minus the drive letter.

Reply:
Looks like there was a `--copy-prefix` feature that actually worked that way, but it was removed ages ago (981473b890912d1dbed58082190723ad6db97501).

Reply:
Hmm thanks! that's a great hint. I've also managed to reproduce it on Linux so hopefully will have a fix today.